testing/testrunner/eventstrace.go

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Elastic-2.0
+
+/*
+ * Copyright 2022 Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under
+ * one or more contributor license agreements. Licensed under the Elastic
+ * License 2.0; you may not use this file except in compliance with the Elastic
+ * License 2.0.
+ */
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os/exec"
+	"time"
+)
+
+type EventsTraceInstance struct {
+	Cmd        *exec.Cmd
+	Stdout     io.ReadCloser
+	Stderr     io.ReadCloser
+	StdoutChan chan string
+	StderrChan chan string
+}
+
+const eventsTraceBinPath = "/EventsTrace"
+
+func (et *EventsTraceInstance) Start() {
+	if err := et.Cmd.Start(); err != nil {
+		fmt.Println("failed to start EventsTrace: ", err)
+		TestFail()
+	}
+
+	et.StdoutChan = make(chan string)
+	go func() {
+		scanner := bufio.NewScanner(et.Stdout)
+		for scanner.Scan() {
+			if err := scanner.Err(); err != nil {
+				fmt.Println("failed to read from EventsTrace stdout: ", err)
+				return
+			}
+			et.StdoutChan <- scanner.Text()
+		}
```

Comment:
The error is only populated _after_ `scanner.Scan()` returns false. So you should check for errors afterwards:
```suggestion
		for scanner.Scan() {
			et.StdoutChan <- scanner.Text()
		}
		if err := scanner.Err(); err != nil {
			fmt.Println("failed to read from EventsTrace stdout: ", err)
			return
		}
```

Reply:
done